project/Deployment.scala

```diff
@@ -13,157 +13,286 @@ import com.typesafe.sbt.SbtNativePackager.Universal
+trait DeploymentType
+case object Local extends DeploymentType
+case object Staging extends DeploymentType
+case object Production extends DeploymentType
+
+class ScastieConfig(val configurationFile: File, val remoteDeploymentFolder: String) {
+  val config = ConfigFactory.parseFile(configurationFile)
+  val userName = "scastie"
+
+  val serverConfig = config.getConfig("com.olegych.scastie.web")
+  val serverHostname = serverConfig.getString("hostname")
+  val serverAkkaPort = serverConfig.getInt("akka-port")
+
+  val metalsConfig = config.getConfig("scastie.metals")
+  val metalsPort = metalsConfig.getInt("port")
+  val cacheExpireInSeconds = metalsConfig.getInt("cache-expire-in-seconds")
+
+  val balancerConfig = config.getConfig("com.olegych.scastie.balancer")
+  val runnersHostname = balancerConfig.getString("remote-hostname")
+  val sbtRunnersPortsStart = balancerConfig.getInt("remote-sbt-ports-start")
+  private val sbtRunnersPortsSize = balancerConfig.getInt("remote-sbt-ports-size")
+  val sbtRunnersPortsEnd = sbtRunnersPortsStart + sbtRunnersPortsSize - 1
+}
+
+object ScastieConfig {
+  def ofType(deploymentType: DeploymentType, deploymentFolder: File): ScastieConfig =
+    deploymentType match {
+      case Local => new ScastieConfig(deploymentFolder / "local.conf", "local")
+      case Staging => new ScastieConfig(deploymentFolder / "staging.conf", "staging")
+      case Production => new ScastieConfig(deploymentFolder / "production.conf", "production")
     }
+
+  def logbackConfigPath(deploymentFolder: File): Path = (deploymentFolder / "logback.xml").toPath
+
 }
 
-class Deployment(rootFolder: File, version: String, sbtDockerImage: ImageName, val logger: Logger) {
-  def deploy(serverZip: Path, metalsServerZip: Path): Unit = {
-    deployRunners()
-    deployServer(serverZip)
-    deployMetalsRunner(metalsServerZip)
+class Deployment(
+  rootFolder: File,
+  version: String,
+  sbtDockerImage: ImageName,
+  metalsDockerImage: ImageName,
+  deploymentType: DeploymentType,
+  val logger: Logger
+) {
+  val deploymentFolder = rootFolder / "deployment"
+  val config = ScastieConfig.ofType(deploymentType, deploymentFolder)
+
+  val sbtDockerNamespace = sbtDockerImage.namespace.get
+  val sbtDockerRepository = sbtDockerImage.repository
+
+  val metalsDockerNamespace = metalsDockerImage.namespace.get
+  val metalsDockerRepository = metalsDockerImage.repository
+
+  def deploy(serverZip: Path, dryRun: Boolean = false) = {
+    val time = LocalDateTime.now()
+    val outputPath = deploymentFolder.toPath.resolve(s"generated-scripts-$time")
+
+    if (!Files.exists(outputPath)) {
+      Files.createDirectory(outputPath)
+    }
+
+    if (createAndVerifyDeploymentScriptsData(outputPath, dryRun)) {
+      // the deployment will be sequential so if anything fails, other services will keep working.
+      deployRunners() && deployMetalsRunner() && deployServer(serverZip)
+    } else {
+      logger.error("Deployment process stopped.")
+    }
```

Comment:
Following on my previous comment, I think this part should be:

```suggestion
    if (dryRun) {
      logger.info("Dry run completed.")
    } else {
      // the deployment will be sequential so if anything fails, other services will keep working.
      createAndVerifyDeploymentScriptsData(outputPath) && deployRunners() && deployMetalsRunner() && deployServer(serverZip)
    }
```